packages/angular_devkit/build_angular/src/builders/browser-esbuild/compiler-plugin.ts

```diff
@@ -291,7 +287,11 @@ export function createCompilerPlugin(
         };
 
         // Create new compilation if first build; otherwise, use existing for rebuilds
-        compilation ??= new AngularCompilation();
+        if (pluginOptions.jit) {
+          compilation ??= new JitCompilation();
+        } else {
+          compilation ??= new AngularCompilation();
```

Comment:
NOTE: `AngularCompilation` will be renamed to `AotCompilation` in a follow-up PR.